threshold.go

```diff
@@ -43,7 +43,7 @@ func (t *threshold) Release() {
 // Wait waits for the counter to drop below the threshold
 func (t *threshold) Wait() {
 	t.mu.Lock()
-	for t.count > t.threshold {
+	for t.count >= t.threshold {
```

Comment:
Are you *sure* this is right? We notify waiters when we see that `threshold` drops *to* count. If we call `Wait` while `threshold` is at `count`, we won't notice when it drops below (unless I'm not thinking about this correctly).

Reply:
It makes it consistent with the comment for `Wait`, doesn't it?

I have a test for this in #5: https://github.com/libp2p/go-libp2p-stream-transport/pull/5/files#diff-3ad278193cb39ee5486b205801d7687aR194.

Reply:
Hm. I appear to be misreading my own code... (`count--` comes *after* the check). You're right.